### PR TITLE
Refactor for testability

### PR DESCRIPTION
**Pull request by iainlane**
This is part of a _PR stack_:

  * #57 <- you are here
  * #58  + #59
  * #60

I started to add a new feature in #60, and wanted to add tests for it. But it turned out to be quite difficult due to the global state we have in the project, and the way the commands are constructed.

So here's a pre-PR with quite a bit of reorganisation of the code. I'll follow up with tests for the CI and PR subcommands, and then the new feature™.

Have tried to split the work into logical commits with each commit having a decent message, but if this is still too big to review then I can push each of those as separate PRs, just let me know. :+1:

### CODE REVIEW

File: cmd/wait-for-github/ci.go [hunk 1]
```diff
@@ -74,108 +74,127 @@ func parseCIArguments(c *cli.Context) error {
...
 	}
 
-	log.Infof("CI is not finished yet, rechecking in %s", cfg.recheckInterval)
+	log.Infof("CI is not finished yet, rechecking in %s", recheckInterval)
 	return nil
 }
 
-func checkCIStatus(c *cli.Context) error {
-	ctx := context.Background()
+type checkSpecificCI struct {
```
Comment (julienduchesne) [inline cmd/wait-for-github/ci.go:103]: `checkSpecificCI`. Not sure about the name. What part is specific?

**Review by julienduchesne — APPROVED** [review]
LGTM. Just some nits

File: cmd/wait-for-github/ci.go (hunk 1 above)
Comment (iainlane) [reply cmd/wait-for-github/ci.go]: It's the `checks` field. When we use this one we call `GetCIStatusForChecks`, it takes a slice of check/status names to look for and makes sure they are all there.

However you made me realise that the structs could be a bit better if the `checks` slice wasn't in the `checkAllCI` one, so I [reorganised them and re-pushed](https://github.com/grafana/wait-for-github/compare/e92072b83a7ff3a36a70bb4b950f944782b27d63..781f8e64edb1ba89fb1fb7bf9dec0939525a967a). Thanks for that :+1:

File: cmd/wait-for-github/ci.go [hunk 2]
```diff
@@ -74,108 +74,127 @@ func parseCIArguments(c *cli.Context) error {
...
+	return ciConfig{
+		owner:  owner,
+		repo:   repo,
+		ref:    ref,
+		checks: c.StringSlice("check"),
+	}, nil
 }
 
-func handleCIStatus(status github.CIStatus) cli.ExitCoder {
+func handleCIStatus(status github.CIStatus, recheckInterval time.Duration) cli.ExitCoder {
```
[the remark this answers is not in the document]
Comment (iainlane) [reply cmd/wait-for-github/ci.go:88]: Yeah. We should pass the logger via the struct or as a parameter or something, and have the config we want to log set as a field on that at a higher level. We could probably set that in the root even, to be honest. I'll look at doing that in a follow-up, cheers.

**Pull request merged**